src/figuresClasses.ts

```
@@ -1,11 +1,81 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+}
+
+export class Triangle implements Figure {
+  readonly shape = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than zero.');
+    }
+
+    const sides = [a, b, c].sort((x, y) => x - y);
+
+    if (sides[2] >= sides[0] + sides[1]) {
+      throw new Error(
+        'The longest side must be less than the sum of the other two sides.',
+      );
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+    const s: number = (a + b + c) / 2;
+    const trArea = Math.sqrt(s * (s - a) * (s - b) * (s - c));
+
+    return Math.floor(trArea * 100) / 100;
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  readonly shape = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+  ) {
+    if (a <= 0) {
+      throw new Error('Resulting value should be even number');
+    }
+  }
+
+  getArea(): number {
+    const circleArea = Math.PI * this.a ** 2;
+
+    return Math.floor(circleArea * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  readonly shape = 'rectangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+  ) {
+    if (a <= 0 || b <= 0) {
+      throw new Error('Resulting value should be even number');
```

The error message is not descriptive enough. According to the task description, the constructor should throw an error if either `a` or `b` is less than or equal to zero. The error message should reflect this condition. See checklist item 3 for more details.